scipy/optimize/_nnls.py

```diff
@@ -136,12 +136,13 @@ def _nnls(A, b, maxiter=None, tol=None):
                      assume_a='sym', check_finite=False)  # B.4
 
         # Inner loop
-        while (iter < maxiter) and (s[P].min() <= tol):  # C.1
+        while (iter < maxiter) and (s[P].min() < 0):  # C.1
             alpha_ind = ((s < tol) & P).nonzero()
             alpha = (x[alpha_ind] / (x[alpha_ind] - s[alpha_ind])).min()  # C.2
+            alpha_ind = ((s < 0) & P).nonzero()
```

Comment:
I think you meant to put this line in place of line 140, is that correct?

Reply:
Yes, I committed a single changed line with git, but it somehow got messed up, probably because of all my debug stuff around that line in my local version. Should have checked that before pushing...